scripts/build/auroTemplateFiller.mjs

```diff
@@ -0,0 +1,200 @@
+// eslint-disable
+import Handlebars from "handlebars"
+import fs from "node:fs/promises";
+
+// declare package.json type with jsdoc
+/**
+ * @typedef {Object} ExamplePackageJson
+ * @property {string} name - Name of the package.
+ * @property {string} version - Version of the package.
+ * @property {Record<string, string>} peerDependencies - Peer dependencies of the package.
+ */
+
+// declare extracted names type with jsdoc
+/**
+ * @typedef {Object} ExtractedNames
+ * @property {string} npm - NPM of the package.
+ * @property {string} namespace - Namespace of the package.
+ * @property {string} namespaceCap - Capitalized namespace of the package.
+ * @property {string} name - Name of the package.
+ * @property {string} nameCap - Capitalized name of the package.
+ * @property {string} version - Version of the package.
+ * @property {string} tokensVersion - Version of the design tokens.
+ * @property {string} wcssVersion - Version of the webcorestylesheets.
+ */
+
+
+export class AuroTemplateFiller {
```

Comment:
That's a good idea! I like this so we can fill template data on any string instead of worrying about how to read/write. Can you give me some callout areas I should look at to make sure I'm covering all my bases?

Reply:
I decided to remove the file handling process but didn't end up consuming it in the AuroTemplateFiller class at all (it doesn't need it). However I AM using the file read/write in the main script for consistent error handling as you recommend!